plugins/CoreAdminHome/Tasks/ArchivesToPurgeDistributedList.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+namespace Piwik\Plugins\CoreAdminHome\Tasks;
+
+use Piwik\Concurrency\DistributedList;
+use Piwik\Date;
+
+/**
+ * Distributed list that holds a list of year-month archive table identifiers (eg, 2015_01 or 2014_11). Each item in the
+ * list is expected to identify a pair of archive tables that contain invalidated archives.
+ *
+ * The archiving purging scheduled task will read items in this list when executing the daily purge.
+ *
+ * This class is necessary in order to keep the archive purging scheduled task fast. W/o a way to keep track of
+ * tables w/ invalid data, the task would have to iterate over every table, which is not desired for a task that
+ * is executed daily.
+ *
+ * If users find other tables contain invalidated archives, they can use the core:purge-old-archive-data command
+ * to manually purge them.
+ */
+class ArchivesToPurgeDistributedList extends DistributedList
+{
+    const OPTION_INVALIDATED_DATES_SITES_TO_PURGE = 'InvalidatedOldReports_DatesWebsiteIds';
+
+    /**
+     * Constructor.
+     */
+    public function __construct()
```

Comment:
No, not useful, just a side-effect of how I code: I put

```
/**
 * TODO
 */
```

over everything when I'm coding, and I fill them out after everything works. I can remove this :)